.github/workflows/gradle.yml

```diff
@@ -1,18 +1,31 @@
-name: Java CI with coverage
+name: BUILD
 
 on:
   push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
       - uses: actions/checkout@v3
 
       - name: Setup Gradle
         uses: gradle/gradle-build-action@v2
+      - name: Run detekt with Gradle Wrapper
+        run: ./gradlew detekt
       - name: Run build with Gradle Wrapper
-        run: gradle detekt
+        run: ./gradlew build
```

Comment:
Pleaase, try to execute 
```bash
./gradlew build -m
```
You will see all the tasks to be executed on `build` task call.

I don't give you the straight answer, just want you to see it by yourself :)

Reply:
done